app/pods/newsletter/index/route.js

```diff
@@ -18,32 +19,51 @@ export default class IndexNewsletterRoute extends Route {
    */
   async model(params) {
     const agenda = this.modelFor('newsletter').agenda; // eslint-disable-line
-    const agendaitems = await this.store
-      .query('agendaitem', {
-        'filter[agenda][:id:]': agenda.id,
-        'filter[show-as-remark]': false,
-        'filter[is-approval]': false,
-        include: 'treatments.newsletter-info',
-        sort: params.sort,
-        'page[size]': PAGE_SIZE.AGENDAITEMS,
-      });
-
-    return Promise.all(agendaitems.map(async(agendaitem) => {
-      const agendaItemTreatments = await agendaitem.get('treatments');
-      const agendaItemTreatment = agendaItemTreatments.firstObject;
-      const newsletterInfo = await agendaItemTreatment.get('newsletterInfo');
-      return {
-        agendaitem,
-        newsletterInfo,
-      };
-    }));
+
+    const filter = {
+      'show-as-remark': false,
+      agenda: {
+        id: agenda.id,
+      },
+      ['is-approval']: false,
+    };
+    let agendaitems = await this.loadAgendaItems(params, filter);
+
+    // KAS-2976 - Old Kaleidos items have undefined is-approval states and need to be queried for this view
+    if (isEmpty(agendaitems)) {
+      filter['is-approval'] = undefined;
+      agendaitems = await this.loadAgendaItems(params, filter);
+    }
```

Comment:
Er zal altijd maar max 1 "is-approval" item zijn.
Kunnen we dan niet gewoon die filter uit de query laten en bij verwerking van de resultaten die eruit filteren als die wordt gevonden?
Als die niet wordt gevonden moet je gewoon niets eruit filteren.
2 keer querien bij oude agendas is niet echt performant (+ cache miss want nieuwe query filter)